src/log/zipkin.cpp

```diff
@@ -14,6 +14,11 @@
 
 namespace fc {
 
+//signal safe
+volatile sig_atomic_t do_sighup = 0;
+//thread safe
+std::atomic<bool> sighup_requested = false;
```

Comment:
Doesn't look like `sighup_requested` is needed as it does the same thing as `do_sighup`

Reply:
I wasn't sure whether `signal-safe` implies `thread-safe`  or not so I used two vars one for signal-safe and the other for thread-safe like the example in https://thomastrapp.com/blog/signal-handler-for-multithreaded-c++.  I will remove `sighup_requested` if you think `do_sighup` is enough.

Reply:
Use https://en.cppreference.com/w/cpp/atomic/atomic_flag for your variable and it will be signal safe and thread atomic OR you can verify `std::atomic<bool>` is lock free.

Reply:
Changed my variable to atomic_flag. Since member func `test()` of atomic_flag is not averrable until c++20, I use `test_and_set()` for checking flag status.

Reply:
I think `std::atomic<bool>` is lock free because `<std::atomic<bool>::is_always_lock_free` equals true.